instance_clusters.go

```diff
@@ -0,0 +1,206 @@
+//
+// Copyright 2020, Serena Fang
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"time"
+)
+
+// InstanceClustersService handles communication with the
+// instance clusters related methods of the GitLab API.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html
+type InstanceClustersService struct {
+	client *Client
+}
+
+// InstanceCluster represents a GitLab Instance Cluster.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/instance_clusters.html
+type InstanceCluster struct {
+	ID                 int                 `json:"id"`
+	Name               string              `json:"name"`
+	Domain             string              `json:"domain"`
+	CreatedAt          *time.Time          `json:"created_at"`
+	ProviderType       string              `json:"provider_type"`
+	PlatformType       string              `json:"platform_type"`
+	EnvironmentScope   string              `json:"environment_scope"`
+	ClusterType        string              `json:"cluster_type"`
+	User               *User               `json:"user"`
+	PlatformKubernetes *PlatformKubernetes `json:"platform_kubernetes"`
+	ManagementProject  *ManagementProject  `json:"management_project"`
+}
+
+func (v InstanceCluster) String() string {
+	return Stringify(v)
+}
+
+// PlatformKubernetes represents a GitLab Instance Cluster PlatformKubernetes.
+type InstanceClusterPlatformKubernetes struct {
```

Comment:
```suggestion
type PlatformKubernetes struct {
```

Reply:
Hi @svanharmelen , thanks for the suggestion. I tried applying it but the type `PlatformKubernetes` is already declared in `https://github.com/xanzy/go-gitlab/blob/master/project_clusters.go:56:6: PlatformKubernetes redeclared in this block`

Reply:
So why not just omit the struct here and use the existing one?

Reply:
Gotcha I see what you mean, pushed these changes!